### PR TITLE
Headless adobe auth validation

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
tested manually against an headless app => works

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/runtime-helpers.js [hunk 1]
```diff
@@ -802,9 +801,61 @@ function setPaths (flags) {
...
+  const ADOBE_HEADLESS_AUTH_ACTION = '/adobeio/shared-validators-v1/headless'
+  const AIO_STATE_KEY = '__aio'
+  const AIO_CONFIG_IMS_ORG_ID = 'project.org.ims_org_id'
+  const AIO_STATE_PUT_ENDPOINT = 'https://adobeio.adobeioruntime.net/api/v1/web/state/put'
+
+  const hasAnAdobeHeadlessAuthSequence = actions.some(a => a.exec && a.exec.kind === 'sequence' && a.exec.components.includes(ADOBE_HEADLESS_AUTH_ACTION))
+  if (hasAnAdobeHeadlessAuthSequence) {
+    // if we use the headless (default auth action) we need to store the ims org id in the
+    // cloud state lib. This is needed by the auth action to perform an org check.
+    const imsOrgId = require('@adobe/aio-lib-core-config').get(AIO_CONFIG_IMS_ORG_ID)
```
Comment (shazron) [inline src/runtime-helpers.js:827]: any advantages here to require inside this scope versus global scope for clarity?

File: src/runtime-helpers.js [hunk 2]
```diff
@@ -802,9 +801,61 @@ function setPaths (flags) {
...
+      },
+      body: JSON.stringify({
+        namespace: owOptions.namespace,
+        key: AIO_STATE_KEY,
+        value: { project: { org: { ims_org_id: imsOrgId } } },
+        ttl: -1 // unlimited
+      })
+    })
+    if (!res.ok) {
+      throw new Error(`failed setting ims_org_id=${imsOrgId} into state lib, received status=${res.status}, please make sure you runtime credentials are correct`)
```
Comment (shazron) [inline src/runtime-helpers.js:846]: spelling: you -> your

Comment (moritzraho) [reply src/runtime-helpers.js:846]: thx!

File: src/runtime-helpers.js (hunk 1 above)
Comment (moritzraho) [reply src/runtime-helpers.js:827]: kept it like this cause this code is not supposed to stay there for long but yeah will move to the global scope

Comment (moritzraho) [reply src/runtime-helpers.js]: ok moved the aio config to global scope but left node fetch require into local scope as it was breaking some tests + node fetch shouldn't be useful in the rt plugin outside of this workaround

**Pull request merged**